src/tools/illink/src/ILLink.RoslynAnalyzer/DynamicallyAccessedMembersAnalyzer.cs

```diff
@@ -76,17 +77,79 @@ public override void Initialize (AnalysisContext context)
 					if (context.OwningSymbol.IsInRequiresUnreferencedCodeAttributeScope (out _))
 						return;
 
-
 					foreach (var operationBlock in context.OperationBlocks) {
 						TrimDataFlowAnalysis trimDataFlowAnalysis = new (context, operationBlock);
 						trimDataFlowAnalysis.InterproceduralAnalyze ();
 						foreach (var diagnostic in trimDataFlowAnalysis.TrimAnalysisPatterns.CollectDiagnostics ())
 							context.ReportDiagnostic (diagnostic);
 					}
 				});
-				context.RegisterSyntaxNodeAction (context => {
-					ProcessGenericParameters (context);
-				}, SyntaxKind.GenericName);
+				// Examine generic instantiations in base types and interface list
+				context.RegisterSymbolAction (context => {
+					var type = (INamedTypeSymbol) context.Symbol;
+					if (type.BaseType is INamedTypeSymbol baseType) {
```

Comment:
What if I put RUC on the type - I don't remember if we decided to still warn on base/interface violation or not.

Reply:
Thanks.